probe-rs/src/session.rs

```diff
@@ -202,6 +202,20 @@ impl Session {
     }
 
     /// Attaches to the core with the given number.
+    ///
+    /// ## Usage  
+    /// Everytime you want to perform an operation on the chip, you need to get the Core handle with the [Session::core() function. This is merely a view into the core.
+    ///
+    /// All the state is actually stored in the Session handle.
+    ///
+    /// The first time you call [Session::core()] for a specific core, it will run the attach/init sequences and return a handle to the core.
+    ///
+    /// Every subsequent call is no-op. It simply returns a convenient handle for the user to use in further operations.
+    ///
+    /// It is strongly advised to never store the core handle for any significant duration! Free it as fast as possible such that other concurrent users have access to the core.
+    ///
+    /// The idea behind this is exactly that: you need the smallest common denominator which you can share between threads and since you sometimes need the core, sometimes the probe or sometimes the target, the session is the only common ground and the only handle you should keep active in your code.
+    ///
```

Comment:
```suggestion
    ///
    /// ## Usage
    /// Everytime you want to perform an operation on the chip, you need to get the Core handle with the [Session::core() method. This [Core] handle is merely a view into the core. And provides a convenient API surface.
    ///
    /// All the state is stored in the [Session] handle.
    ///
    /// The first time you call [Session::core()] for a specific core, it will run the attach/init sequences and return a handle to the [Core].
    ///
    /// Every subsequent call is a no-op. It simply returns the handle for the user to use in further operations without calling any int sequences again.
    ///
    /// It is strongly advised to never store the [Core] handle for any significant duration! Free it as fast as possible such that other stakeholders can have access to the [Core] too.
    ///
    /// The idea behind this is: You need the smallest common denominator which you can share between threads. Since you sometimes need the [Core], sometimes the [Probe] or sometimes the [Target], the [Session] is the only common ground and the only handle you should actively store in your code.
    ///
```